### ISSUE TITLE
`as const` cast wrong colour when using typescript 4.5.2

### ISSUE THREAD

**Issue by Nokel81** [microsoft/vscode#138362]
Does this issue occur when all extensions are disabled?: Yes

Version: 1.62.3
Commit: ccbaa2d27e38e5afa3e5c21c1c7bef4657064247
Date: 2021-11-17T07:59:13.865Z
Electron: 13.5.2
Chrome: 91.0.4472.164
Node.js: 14.16.0
V8: 9.1.269.39-electron.0
OS: Darwin x64 19.6.0

Steps to Reproduce:

1. Create a typescript file with the line `[1, "hello"] as const`
2. Notice that `const` is green (as in a type) instead of blue since it is a keyword.

**Comment by mjbvz** [comment]
Does this reproduce in the [latest VS Code insiders build](https://code.visualstudio.com/insiders/) with all [extensions disabled](https://github.com/microsoft/vscode/wiki/Performance-Issues#run-with-extensions-disabled)?

**Comment by Nokel81** [comment]
@mjbvz Nope, still a problem. I see that it is basically the same as https://github.com/microsoft/TypeScript/issues/46998
